### ISSUE TITLE
Refactor to use Unreal Engine 4

### ISSUE THREAD

**Issue by athix** [AeonSplice/aeonsplice#19]
Oh god oh god, the world's on fire. Also public c++ libraries are rather lacking.

**Comment by fire** [comment]
I offer my help in Unreal Engine, but I can also disappear if you hate me :(.

**Comment by athix** [comment]
I'll gladly accept any help I can get. Not really sure what happened between us, horrible memory and all. How about a fresh start?

**Comment by fire** [comment]
Sure, what's the game idea. What do you want to do?

**Comment by athix** [comment]
For the high level concept, I want to mesh the feeling of large fleet battles (e.g. Battlestar Galactica) with the tactical mindset of games like Chess and Feudal. Also, I want to use the project as a learning experience for physics and high-level mathematics. (Model everything as accurately as possible, and use continuous collision that takes into account object's properties rather than discrete hard-body physics.)

I'm still on the fence on 2d vs 3d, and most of the game mechanics I worked out are in my file archives somewhere. I'll be spending some time digging them up and putting it all on the wiki.

The biggest problem I have is that Unreal Engine 4 (and Unity, but UE4 seemed more appealing) is a big magic black box. GLFW / similar was nice because of the transparency, but I'd rather spend my time fleshing out game mechanics than a game engine. That being said, everything I've found so far is geared towards pumping out either a CoD clone, platformer, or RTS. The idea of bending UE4 to do my TBS seems daunting, to say the least.
